lib/engine/part/city.rb

```diff
@@ -47,8 +47,12 @@ def tokened?
         !@tokens.compact.empty?
       end
 
-      def tokened_by?(corporation)
-        @tokens.any? { |t| t&.corporation == corporation } || @extra_tokens.any? { |t| t&.corporation == corporation }
+      def tokened_by?(corporation, types: [])
+        token = (@tokens + @extra_tokens).find { |t| t&.corporation == corporation }
+        return false unless token
+        return true if types == []
+
+        types.include?(token.type)
```

Comment:
Do we need to consider a game where a corporation could have both a normal station token and another type of token in the same city?